crates/openid4vci/src/types/credential.rs

```diff
@@ -64,7 +78,7 @@ pub enum CredentialFormatProfile {
     LdpVc {
         /// JSON array as defined in [VC_DATA](https://www.w3.org/TR/vc-data-model/), Section 4.1.
         #[serde(rename = "@context")]
-        context: Vec<String>,
+        context: LinkedDataContext,
```

Comment:
> The value of the @context [property](https://www.w3.org/TR/vc-data-model/#dfn-property) MUST be an ordered set where the first item is a [URI](https://www.w3.org/TR/vc-data-model/#dfn-uri) with the value https://www.w3.org/2018/credentials/v1. For reference, a copy of the base context is provided in Appendix [B.1 Base Context](https://www.w3.org/TR/vc-data-model/#base-context). Subsequent items in the array MUST express context information and be composed of any combination of [URIs](https://www.w3.org/TR/vc-data-model/#dfn-uri) or objects


So I think this should be a `List<String | Map>` (or we could even go further and check that the first url is vc, but that's not really necessary I think)

Reply:
Couldn't we require a list that takes either string or map instead of adding another wrapper enum. It makes this code harded to call, which we could maybe just guard using normal types?

Reply:
No in Rust you cannot have an OR type and this is the common way of using different types for a single field.

Reply:
Will update with the new context. I checked the `json-ld` specification and that was a bit more loose.

Reply:
Should be fixed now.